zerver/views/video_calls.py

```diff
@@ -167,22 +168,12 @@ def get_bigbluebutton_url(request: HttpRequest, user_profile: UserProfile,
     # https://docs.bigbluebutton.org/dev/api.html#create for reference on the API calls
     # https://docs.bigbluebutton.org/dev/api.html#usage for reference for checksum
     id = "zulip-" + str(random.randint(100000000000, 999999999999))
-    password = b32encode(secrets.token_bytes(7))[:10].decode()
-    create_params = urlencode({  # type: ignore[type-var] # https://github.com/python/typeshed/issues/4234
-        "meetingID": id,
-        "name": meeting_name,
-        "moderatorPW": password,
-        # We generate the password this way to make sure only the server knows it.
-        # This is a easy way to prevent the user to get the data of the shared url
-        # and use it later and skipping Zulip validation
-        "attendeePW": hashlib.sha1((password + settings.BIG_BLUE_BUTTON_SECRET).encode()).hexdigest(),
-    }, quote_via=quote)
-    checksum = hashlib.sha1(("create" + create_params + settings.BIG_BLUE_BUTTON_SECRET).encode()).hexdigest()
+    password = b32encode(secrets.token_bytes(7))[:20].decode()
+    signer = Signer()
     url = add_query_to_redirect_url("/calls/bigbluebutton/join", urlencode({  # type: ignore[type-var] # https://github.com/python/typeshed/issues/4234
-        "meeting_id": id,
-        "name": meeting_name,
-        "password": password,
-        "checksum": checksum
+        "meeting_id": signer.sign(id),
+        "name": signer.sign(meeting_name),
+        "password": signer.sign(password)
```

Comment:
Hmm, wouldn't it be simpler to use signer.dumps to sign an entire dict instead of signing element by element?

Reply:
As these are parameters in an URL I do not think dumps would work.
To use dumps we would have to transfer the whole thing as one JSON parameter.
I personally do think that this (currently in the PR) is a more clear approach, but are open to discuss.